### PR TITLE
Task4 - Creating SQL DB

### PR DESCRIPTION
**Pull request by tensegrity666**
## Selfgrade:
Reviewers should verify the lambda functions by invoking them through provided URLs.

- [x] 1 - Task 4.1 is implemented

- [x] 3 - TASK 4.2 is implemented lambda links are provided and returns data

- [x] 4 - TASK 4.3 is implemented lambda links are provided and products is stored in DB (call TASK 4.2 to see the product)
- [x] 5 - Your own Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. Link to a working Front-End application is provided for cross-check reviewer.

Additional (optional) tasks (but nice to have):
- [x] +1 - POST/products lambda functions returns error 400 status code if product data is invalid
- [x] +1 - All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
- [x] +1 - All lambdas do console.log for each incoming requests and their arguments
- [ ] +1 - Transaction based creation of product (in case stock creation is failed then related to this stock product is not created and not ready to be used by the end user and vice versa) (https://devcenter.kinvey.com/nodejs/tutorials/bl-transactional-support)
## Total: 8
- - - 
_Write SQL script to fill tables with test examples. Store it in your GIT repository. Execute it for your DB to fill data._
Examples contains in `./product-service/sql/`
- - -
I tested creating a new post using the POST method using the Lambda interface:
https://yadi.sk/i/zNESJ6x5P3RAIg
using POSTMAN, and using frontend interface

__Product schema:__<br>
```

  {
    "title": text,
    "description": text,
    "picture": text,
    "price": int
  }

```
- - -
- - -
### Frontend ([link to fe-repo](https://github.com/tensegrity666/nodejs-aws-fe)):
https://d4b3oofc3xqi9.cloudfront.net

### Backend (branch __dev__)
* API get all products:
https://jqnaa7z9ua.execute-api.eu-west-1.amazonaws.com/dev/product

* API get product by id (example with one of id's):
https://jqnaa7z9ua.execute-api.eu-west-1.amazonaws.com/dev/product/c528cb4b-b705-4ed2-b15c-a0355bb5eaf1
- - - 

## [SWAGGER](https://app.swaggerhub.com/apis/tensegrity666/product-service/1.0.0)

### CODE REVIEW

**Review by IDiamondDragon — COMMENTED** [review]
When i go to this link https://jqnaa7z9ua.execute-api.eu-west-1.amazonaws.com/dev/product/c528cb4b-b705-4ed2-b15c-a0355bb5eaf1 I have got not one product but two object 

[
[
{
"id": "c528cb4b-b705-4ed2-b15c-a0355bb5eaf1",
"title": "Fox",
"description": "Short Product Description 2",
"picture": "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQhOVE6UDSrtvlD46SF1MB-k-h1177aASQhZg&usqp=CAU",
"price": 49
}
],
[
{
"id": "1368f59e-3fac-450d-9146-f6c07b153aae",
"list_id": "c528cb4b-b705-4ed2-b15c-a0355bb5eaf1",
"count": 7
}
]
]

but we need to get one object

*FE: One product model as a result of BE models join(product and it's stock) *

  Product model example on Frontend side: 

  {
    id: '19ba3d6a-f8ed-491b-a192-0a33b71b38c4',
    count: 2
    price: 200,
    title: ‘Product Title’,
    description: ‘This product ...’
  }

Something like this https://584prtl1u5.execute-api.eu-west-1.amazonaws.com/dev/products/a731e308-a346-4607-a8fc-724efeb68502

File: product-service/src/postgress-handlers/get-by-id.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Client } from 'pg';
+import config from '../common/pg-config';
+
+const getProductById = async (event) => {
+  const client = new Client(config);
+  await client.connect();
+
+  try {
+    const { rows: product } = await client.query(`select * from products
+    where id = '${event.pathParameters.id}'`);
```
Comment (Anisotropic) [inline product-service/src/postgress-handlers/get-by-id.js:10]: Where is JOIN 'stocks' table to read "counts"?

**Comment by LittleCryingGirl** [comment]
I suggest that 'Your own Frontend application is integrated with product service' means not only get request at main page, but also page
https://d4b3oofc3xqi9.cloudfront.net/admin/products

and after new product created on https://d4b3oofc3xqi9.cloudfront.net/admin/product-form app should successfully request all products after redirect

I think its quite incorrect that https://jqnaa7z9ua.execute-api.eu-west-1.amazonaws.com/dev/product/c528cb4b-b705-4ed2-b15c-a0355bb5eaf1 returns array with 2 object and its both different entities

**Review by LittleCryingGirl — CHANGES_REQUESTED** [review]
need to add count field
need to implement admin/products page

**Pull request merged**